WordPress/gutenberg
# Blocks: Avoid separate host matching constant for embeds

This pull request seeks to refactor how we consider non-previewable hosts to avoid the need to maintain a list of host names which are separate from the settings definitions of the blocks on which they act. The changes here assign `previewable` as a property of the Facebook and SmugMug embed blocks. This also helps reduce redundancy between the host names and pattern matching of a block, since the former had forced the implementation of hostname normalization which otherwise wouldn't be necessary when relying on the existing patterns matching. It is also consistent with other embed-specific behaviors (e.g. `responsive`) being assigned in `core-embeds.js`.

An interesting consequence of the fact that these were maintained separately is that the pattern for SmugMug URLs was inaccurate since it didn't capture subdomains (and likewise did not match the [equivalent pattern defined in PHP](https://github.com/WordPress/wordpress-develop/blob/98bee9bf081e2c43c8908002d75870c3dfd61b19/src/wp-includes/class-oembed.php#L61)). Thus, it was not being transformed to the "SmugMug" embed variant. It became obvious there was an issue when testing a SmugMug URL had initially regressed the issue described in #11960. This was then fixed in ea10e03.

**Testing instructions:**

Verify that embeds preview or don't preview, depending on whether the embed type can support previews:

Previewable example:

- https://twitter.com/CincinnatiZoo/status/1050770683536568322

Non-previewable examples:

- https://www.facebook.com/SharingCourage/posts/331841743591602
- https://johndavis.smugmug.com/Nature/Landscape/i-NntLPZ8
